test/integration/developer_portal/passwords_controller_test.rb

```diff
@@ -59,6 +69,37 @@ def test_update_password
     assert_not body.include? 'g-recaptcha'
   end
 
+  test 'create responds with error message when detects spam' do
+    provider.settings.update_attributes(spam_protection_level: :auto)
+    ThreeScale::SpamProtection::Protector.any_instance.stubs(spam?: true)
+    Recaptcha::Verify.stubs(skip?: false)
+
+    post developer_portal.admin_account_password_path(email: 'user@example.com')
+    assert_equal 'Spam protection failed.', flash[:error]
+    assert_redirected_to developer_portal.new_admin_account_password_path(request_password_reset: true)
+  end
+
+  test 'create sends the email when captcha passes and finds the email' do
+    provider.settings.update_attributes(spam_protection_level: :auto)
+    ThreeScale::SpamProtection::Protector.any_instance.stubs(spam?: true)
+    Recaptcha::Verify.stubs(skip?: true)
+
+    post developer_portal.admin_account_password_path(email: user.email)
+    assert_equal 'A password reset link has been emailed to you.', flash[:notice]
+    assert_redirected_to developer_portal.login_path
+    assert user.reload.lost_password_token
```

Comment:
Asserting something is defined could be dangerous (maybe here, it isn't). Would it be a good idea/possible to stub the token and assert it is exactly the one we expect? Something like:
```
post developer_portal.admin_account_password_path(email: user.email, token: 123)
...
assert_equal user.reload.lost_password_token, 123

Reply:
Unfortunately, I cannot check for any specific value because it is always randomly generated:
https://github.com/3scale/porta/blob/7645462e3696bcdd2b05704573c23ea344bded1c/app/models/user.rb#L234-L252

However, I still can improve the test by ensuring:
1. The lost_password_token does not only contain a value at this point in time, but also it does not contain it before the request
2. The lost_password_token_generated_at has the current date
3. It also sends the email 👀